openlibrary/macros/ReadingLogButton.html

```diff
@@ -1,13 +1,26 @@
-$def with (work, read_status)
+$def with (work, read_status, url=None, searchPage = None)
 
-<form method="POST" action="$(work.key)/bookshelves.json?debug=true" class="reading-log-lite">
-  <select name="bookshelf_id">
-    $ shelf_names = {1: _('Want to Read'), 2: _('Currently Reading'), 3: _('Already Read') }
-    $for shelf_id in [1, 2, 3]:
-      $ is_selected = read_status == shelf_id
-      <option value="$shelf_id" $cond(is_selected, 'selected')>
-        $cond(is_selected, '✓') $shelf_names[shelf_id]
-      </option>
-    <option value="-1">$_('Remove')</option>
-  </select>
-</form>
+$if ctx.user:
+  <form method="POST" action="$(work.key)/bookshelves.json?debug=true" class="reading-log-lite $(searchPage)">
+    <select name="bookshelf_id">
+      $ shelf_names = {1: _('Want to Read'), 2: _('Currently Reading'), 3: _('Already Read') }
+      $for shelf_id in [1, 2, 3]:
+        $ is_selected = read_status == shelf_id
+        <option value="$shelf_id" $cond(is_selected, 'selected')>
+          $cond(is_selected, '✓') $shelf_names[shelf_id]
+        </option>
+      <option value="-1">$_('Remove')</option>
+    </select>
+  </form>
+
+$else:
+  <div class="dropit">
+    <div class="dropper on">
+        <div class="log-work">
+            <button class="unactivated" type="submit">$_('Want to Read')</button>
```

Comment:
I think that this should be wrapped in a form, similar to the dropper on in the `/lists/widget.html` template.  This will ensure that patrons are directed to the login page when they press the button directly.

Login redirection will be handled by the bookshelf endpoint when this is inside of a form.